### PR TITLE
1248 test case myidentityscreen

### LINKED ISSUE #1248: Test case: MyIdentityScreen

### PR DESCRIPTION
**Pull request by Ekaanth**

### CODE REVIEW

File: src/ui/screens/MyIdentityScreen/ManageIdentityScreen.test.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    address: '14yx4vPAACZRhoDQm1dyvXD3QdRQyCRRCe5tj1zPomhhS29a',
+  },
+} as RouteProp<AccountsStackParamList, typeof manageIdentityScreen>;
+
+const openURLSpy = jest.spyOn(Linking, 'openURL');
+
+describe('ManageIdentityScreen', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
```
Comment (ishrak-imam) [inline src/ui/screens/MyIdentityScreen/ManageIdentityScreen.test.tsx:33]: Is it needed?

File: src/ui/screens/MyIdentityScreen/ManageIdentityScreen.test.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    await findByText('View externally');
+  });
+
+  const identities = [
+    ['@nachortti', 'https://twitter.com/@nachortti'],
+    ['@raul.rtti:matrix.parity.io', 'https://matrix.to/#/@raul.rtti:matrix.parity.io'],
+    ['www.nachortti.com', 'www.nachortti.com'],
+  ];
+
+  it.each(identities)('should click on %s to navigate to %s url', async (id, url) => {
```
Comment (silva-fj) [inline src/ui/screens/MyIdentityScreen/ManageIdentityScreen.test.tsx:62]: `it('should linkify linkable identities' ...`

Let's include all the identities and assert as well that when pressed, `openUrl` should not be called

File: src/ui/screens/MyIdentityScreen/ManageIdentityScreen.test.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    const alertSpy = jest.spyOn(Alert, 'alert');
+    const {findByText} = render(<ManageIdentityScreen navigation={navigation} route={route} />);
+    fireEvent.press(await findByText('Clear Identity'));
+    expect(alertSpy).toBeCalledTimes(1);
+    waitFor(async () => {
+      await findByText('Clear Identity');
+    });
+  });
+
+  it('should navigate to sub-identities on click of Set Sub-identities', async () => {
```
Comment (silva-fj) [inline src/ui/screens/MyIdentityScreen/ManageIdentityScreen.test.tsx]:
```suggestion
  it('should navigate to sub-identities when tapping on Set Sub-identities', async () => {
```

**Comment by ishrak-imam** [comment]
MyIdentity screen has a bug in the submit identity form. I will be fixing that first to make the tests work again.

**Review by ishrak-imam — APPROVED** [review]

**Comment by Ekaanth** [comment]
> MyIdentity screen has a bug in the submit identity form. I will be fixing that first to make the tests work again.

@ishrak-imam  The test cases seem to work fine now. I don't see any CI error.

**Comment by ishrak-imam** [comment]
> > MyIdentity screen has a bug in the submit identity form. I will be fixing that first to make the tests work again.
> 
> @ishrak-imam The test cases seem to work fine now. I don't see any CI error.

Let's get this merged then.

**Pull request merged**